EmailService not working due to unsatisfied reference to MailTemplateManager in AEM on prem

## Required Information

* [ ] AEM Version, including Service Packs, Cumulative Fix Packs, etc: 6.5.21
* [ ] ACS AEM Commons Version: 6.9.0+
* [ ] Reproducible on Latest? yes


### Expected Behavior

ACS Commons EmailService works in AEM on prem.

### Actual Behavior

With https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/3422 (ACS Commons 6.9.0) MailTemplateManager was introduced and added as a reference to EmailService and the implementation for the MailTemplateManager was placed in bundle-cloud and bundle-onprem. Unfortunately the bundle-onprem is missing in acs-aem-commons-all-6.9.*.zip files. Looks like the dependency entry for bundle-onprem is missing in all/pom.xml.
I did a quick try and after adding the dependency to all/pom.xml bundle-onprem was part of the resulting zip file but the build failed with:
`[ERROR] [artifact-rules] com.adobe.acs:acs-aem-commons-bundle-onprem:6.9.5-SNAPSHOT: The Adobe ACS Commons onprem bundle is not supported (adobe/consulting:acs-aem-commons-all:6.9.5-SNAPSHOT)`

In addition to the missing bundle in the zip file it looks like something with the implementation is not working. I manually installed a locally build acs-aem-commons-bundle-onprem:6.9.4 but then the following error occurred:
`java.lang.ClassNotFoundException: com.adobe.acs.commons.email.impl.MailTemplateManager not found by acs-aem-commons-bundle-onprem`

### Steps to Reproduce

For the first part of the issue AEM is not needed to reproduce this. Just open acs-aem-commons-all-6.9.*.zip and go to jcr_root\apps\acs-commons\install. There only acs-aem-commons-bundle is present and acs-aem-commons-bundle-onprem is missing. 
With AEM navigate to http://localhost:4502/system/console/components and search for com.adobe.acs.commons.email.impl.EmailServiceImpl. The status will be "unsatisfied (reference)"

For the second part navigate to http://localhost:4502/system/console/components and search for com.adobe.acs.commons.email.impl.MailTemplateManagerImpl. The status will be "satisfied" and in the error.log of AEM there will be messages like:
`java.lang.ClassNotFoundException: com.adobe.acs.commons.email.impl.MailTemplateManager not found by acs-aem-commons-bundle-onprem`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 
 - #3494 - Remove offline instrumentation with Jacoco
 
+### Fixed
+
+- #3471 - EmailService not working due to unsatisfied reference to MailTemplateManager in AEM on prem
+
 ## 6.9.10 - 2024-12-13
 
 ### Added 

--- a/all/pom.xml
+++ b/all/pom.xml
@@ -48,6 +48,7 @@
                 <configuration>
                     <packageType>container</packageType>
                     <excludes>**/META-INF/*,**/thumbnail.png</excludes>
+                    <embeddedTarget>/apps/acs-commons/install</embeddedTarget>
                 </configuration>
                 <executions>
                     <execution>
@@ -59,40 +60,14 @@
                             <name>acs-aem-commons-all</name>
                             <embeddeds>
                                 <embedded>
-                                    <groupId>${project.groupId}</groupId>
-                                    <artifactId>acs-aem-commons-bundle</artifactId>
-                                    <target>/apps/acs-commons/install</target>
-                                    <filter>true</filter>
-                                    <isAllVersionsFilter>true</isAllVersionsFilter>
-                                </embedded>
-                                <embedded>
-                                    <groupId>${project.groupId}</groupId>
-                                    <artifactId>acs-aem-commons-bundle-onprem</artifactId>
-                                    <target>/apps/acs-commons/install</target>
-                                    <filter>true</filter>
-                                    <isAllVersionsFilter>true</isAllVersionsFilter>
-                                </embedded>
-                                <embedded>
-                                    <groupId>${project.groupId}</groupId>
-                                    <artifactId>acs-aem-commons-ui.apps</artifactId>
+                                    <!-- exclude cloud specific dependencies -->
+                                    <artifactId>~acs-aem-commons-bundle-cloud</artifactId>
                                     <classifier>~cloud</classifier>
-                                    <target>/apps/acs-commons/install</target>
+                                    <type>jar,zip</type>
                                     <filter>true</filter>
                                     <isAllVersionsFilter>true</isAllVersionsFilter>
-                                </embedded>
-                                <embedded>
-                                    <groupId>${project.groupId}</groupId>
-                                    <artifactId>acs-aem-commons-ui.content</artifactId>
-                                    <target>/apps/acs-commons/install</target>
-                                    <filter>true</filter>
-                                    <isAllVersionsFilter>true</isAllVersionsFilter>
-                                </embedded>
-                                <embedded>
-                                    <groupId>${project.groupId}</groupId>
-                                    <artifactId>acs-aem-commons-ui.config</artifactId>
-                                    <target>/apps/acs-commons/install</target>
-                                    <filter>true</filter>
-                                    <isAllVersionsFilter>true</isAllVersionsFilter>
+                                    <!-- only consider direct dependencies -->
+                                    <excludeTransitive>true</excludeTransitive>
                                 </embedded>
                             </embeddeds>
                         </configuration>
@@ -206,10 +181,29 @@
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>
+            <artifactId>acs-aem-commons-bundle-onprem</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.adobe.acs</groupId>
+            <artifactId>acs-aem-commons-bundle-cloud</artifactId>
+            <version>${project.version}</version>
+            <type>jar</type>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
             <artifactId>acs-aem-commons-ui.apps</artifactId>
             <version>${project.version}</version>
             <type>zip</type>
             <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.adobe.acs</groupId>
+            <artifactId>acs-aem-commons-ui.apps</artifactId>
+            <classifier>cloud</classifier>
+            <version>${project.version}</version>
+            <type>zip</type>
         </dependency>
         <dependency>
             <groupId>${project.groupId}</groupId>
@@ -278,39 +272,17 @@
                                         <cloudManagerTarget>all</cloudManagerTarget>
                                     </properties>
                                     <embeddeds>
+                                        <!-- exclude onprem/classic specific artifacts-->
                                         <embedded>
-                                            <groupId>${project.groupId}</groupId>
-                                            <artifactId>acs-aem-commons-bundle</artifactId>
-                                            <target>/apps/acs-commons/install</target>
+                                            <artifactId>~acs-aem-commons-bundle-onprem,~acs-aem-commons-ui.apps</artifactId>
                                             <filter>true</filter>
                                             <isAllVersionsFilter>true</isAllVersionsFilter>
+                                            <!-- only consider direct dependencies -->
+                                            <excludeTransitive>true</excludeTransitive>
                                         </embedded>
                                         <embedded>
-                                            <groupId>${project.groupId}</groupId>
-                                            <artifactId>acs-aem-commons-bundle-cloud</artifactId>
-                                            <target>/apps/acs-commons/install</target>
-                                            <filter>true</filter>
-                                            <isAllVersionsFilter>true</isAllVersionsFilter>
-                                        </embedded>
-                                        <embedded>
-                                            <groupId>${project.groupId}</groupId>
                                             <artifactId>acs-aem-commons-ui.apps</artifactId>
                                             <classifier>cloud</classifier>
-                                            <target>/apps/acs-commons/install</target>
-                                            <filter>true</filter>
-                                            <isAllVersionsFilter>true</isAllVersionsFilter>
-                                        </embedded>
-                                        <embedded>
-                                            <groupId>${project.groupId}</groupId>
-                                            <artifactId>acs-aem-commons-ui.content</artifactId>
-                                            <target>/apps/acs-commons/install</target>
-                                            <filter>true</filter>
-                                            <isAllVersionsFilter>true</isAllVersionsFilter>
-                                        </embedded>
-                                        <embedded>
-                                            <groupId>${project.groupId}</groupId>
-                                            <artifactId>acs-aem-commons-ui.config</artifactId>
-                                            <target>/apps/acs-commons/install</target>
                                             <filter>true</filter>
                                             <isAllVersionsFilter>true</isAllVersionsFilter>
                                         </embedded>
@@ -348,26 +320,14 @@
                                 <goals>
                                     <goal>project-analyse</goal>
                                 </goals>
+                                <configuration>
+                                    <classifier>cloud</classifier>
+                                </configuration>
                             </execution>
                         </executions>
                     </plugin>
                 </plugins>
             </build>
-            <dependencies>
-                <dependency>
-                    <groupId>com.adobe.acs</groupId>
-                    <artifactId>acs-aem-commons-bundle-cloud</artifactId>
-                    <version>${project.version}</version>
-                    <type>jar</type>
-                </dependency>
-                <dependency>
-                    <groupId>com.adobe.acs</groupId>
-                    <artifactId>acs-aem-commons-ui.apps</artifactId>
-                    <classifier>cloud</classifier>
-                    <version>${project.version}</version>
-                    <type>zip</type>
-                </dependency>
-            </dependencies>
         </profile>
     </profiles>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -219,14 +219,8 @@
                     <groupId>biz.aQute.bnd</groupId>
                     <artifactId>bnd-maven-plugin</artifactId>
                     <version>${bnd.version}</version>
-                    <executions>
-                        <execution>
-                            <id>bnd-process</id>
-                            <goals>
-                                <goal>bnd-process</goal>
-                            </goals>
-                            <configuration>
-                                <bnd><![CDATA[
+                    <configuration>
+                        <bnd><![CDATA[
 # a lot of bundle header are generated from pom elements by default: https://github.com/bndtools/bnd/tree/master/maven/bnd-maven-plugin#default-bundle-headers
 Bundle-Category: acs-aem-commons
 # export all versioned packages except for conditional ones (https://github.com/bndtools/bnd/issues/3721#issuecomment-579026778)
@@ -248,8 +242,14 @@ Bundle-DocURL: https://adobe-consulting-services.github.io/acs-aem-commons/
 # support only DS 1.4 (https://github.com/bndtools/bnd/pull/3121/files)
 -dsannotations-options: version;maximum=1.4.0,inherit
 -metatypeannotations-options: version;maximum=1.4.0
-                                ]]></bnd>
-                            </configuration>
+                        ]]></bnd>
+                    </configuration>
+                    <executions>
+                        <execution>
+                            <id>bnd-process</id>
+                            <goals>
+                                <goal>bnd-process</goal>
+                            </goals>
                         </execution>
                     </executions>
                     <dependencies>

